onnxruntime/core/providers/cuda/tensor/expand_impl.cu

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "expand_impl.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T, int pad_mode>
+__global__ void _ExpandKernel(
```

Comment:
> _Ex [](start = 15, length = 4)

Why do we choose to name kernels with leading undersore? Global names with leading underscores are reserved by C++ standard and are not recommended. @Ke ?

Reply:
This is Python style of private function, please change it to conform to C++ standard.

---
In reply to: [297304474](https://github.com/microsoft/onnxruntime/pull/1292#discussion_r297304474) [](ancestors = 297304474)

Reply:
This seems to be a CUDA pattern. All the other operators use the same pattern of __global__. See this as an example.

https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cuda/tensor/pad_impl.cu#L18

Reply:
I know it is a pattern. It is a wrong pattern.

---
In reply to: [297407942](https://github.com/microsoft/onnxruntime/pull/1292#discussion_r297407942) [](ancestors = 297407942)

Reply:
Should this pattern be updated in a separate PR? It seems like a sweeping change.

---
In reply to: [297441897](https://github.com/microsoft/onnxruntime/pull/1292#discussion_r297441897) [](ancestors = 297441897,297407942)

Reply:
You may use the right pattern in this PR. A sweep of all other kernels can come later.

---
In reply to: [297455917](https://github.com/microsoft/onnxruntime/pull/1292#discussion_r297455917) [](ancestors = 297455917,297441897,297407942)

Reply:
> You may use the right pattern in this PR. A sweep of all other kernels can come later.
> 
> In reply to: [297455917](https://github.com/microsoft/onnxruntime/pull/1292#discussion_r297455917) [](ancestors = 297455917,297441897,297407942)

Updated the kernel name and removed leading underscore.